cloudsql/mysql/database-sql/README.md

```diff
@@ -0,0 +1,107 @@
+# Connecting to Cloud SQL (MySQL) from a Go web app
+
+This repo contains the Go source code for a simple web app that can be deployed to App Engine Standard. It is a demonstration of how to connect to a MySQL instance in Cloud SQL. The application is the "Tabs vs Spaces" web app used in the [Building Stateful Applications With Kubernetes and Cloud SQL](https://www.youtube.com/watch?v=qVgzP3PsXFw&t=1833s) session at Cloud Next '19.
+
+## Before you begin
+
+1. If you haven't already, set up a Go Development Environment by following the [Go setup guide](https://cloud.google.com/go/docs/setup) and 
+[create a project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project).
+
+1. Create a 2nd Gen Cloud SQL Instance by following these 
+[instructions](https://cloud.google.com/sql/docs/mysql/create-instance). Note the connection string,
+database user, and database password that you create.
+
+1. Create a database for your application by following these 
+[instructions](https://cloud.google.com/sql/docs/mysql/create-manage-databases). Note the database
+name. 
+
+1. Create a service account with the 'Cloud SQL Client' permissions by following these 
+[instructions](https://cloud.google.com/sql/docs/mysql/connect-external-app#4_if_required_by_your_authentication_method_create_a_service_account).
+Download a JSON key to use to authenticate your connection. 
+
+1. Use the information noted in the previous steps to set environment variables for local development and testing. Follow the instructions below for your local operating system. Note: Saving credentials in environment variables is convenient, but not secure - consider a more
+secure solution such as [Cloud KMS](https://cloud.google.com/kms/) to help keep secrets safe.
+
+### Linux / Mac OS
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service/account/key.json
+export INSTANCE_CONNECTION_NAME='<MY-PROJECT>:<INSTANCE-REGION>:<INSTANCE-NAME>'
+export DB_USER='<DB_USER_NAME>'
+export DB_PASS='<DB_PASSWORD>'
+export DB_NAME='<DB_NAME>'
+```
+
+### Windows/PowerShell
+The Cloud SQL Proxy currently requires the use of TCP on Windows. The optional ```DB_TCP_HOST``` environment variable can be used to specify the proxy's local address, and the ```INSTANCE_CONNECTION_NAME``` environment variable is not needed in this case.
+```powershell
+$env:GOOGLE_APPLICATION_CREDENTIALS="<CREDENTIALS_JSON_FILE>"
+$env:DB_TCP_HOST="127.0.0.1:3306"
+$env:DB_USER="<DB_USER_NAME>"
+$env:DB_PASS="<DB_PASSWORD>"
+$env:DB_NAME="<DB_NAME>"
+```
+
+## Running locally
+
+To run this application locally, download and install the `cloud_sql_proxy` by
+following the instructions
+[here](https://cloud.google.com/sql/docs/mysql/sql-proxy#install). Once the
+proxy has been downloaded, use the following commands start it running in the background for local testing on either Linux/Mac OS or Windows.
+
+### Launch the proxy on Linux/Mac OS
```

Comment:
Same with TCP instructions here. 

Since everything but the command is the same between them, maybe it would make more sense to do seperate into "Launch proxy with TCP" and "Launch proxy with Unix Domain Sockets" and include the commands for each in the individual sections

Reply:
Yes, agree that makes more sense now that we have the DB_TCP_HOST option. I'll refactor these instructions into "Launch proxy with TCP" and "Launch proxy with Unix Domain Sockets" sections.